relationships/synthesis/INFRA-VSPHEREVM-to-INFRA-HOST.yml

```diff
@@ -0,0 +1,30 @@
+relationships:
+  - name: vsphereVMContainsHost
+    version: "1"
+    origins:
+      - OnHost Integration
+    conditions:
+      - attribute: eventType
+        anyOf: ["VSphereVmSample"]
+      - attribute: vmHostname
+        present: true
+      # This additional attribute is required because the relationship doesn't apply to every environment
+      # (the VSPHEREVM vmHostname and the HOST hostname don't match in every scenario).
+      # Adding this label to VSphereVmSamples means that a relationship between the VSPHEREVM entity and the
+      # HOST entity whose hostname matches the vmHostname is expected.
+      - attribute: label.enableVMHostNameRelationship
```

Comment:
Can we also add additional information about this attribute? For example, we might want to explain the billing implications in case other customers see this rule and decide to implement this hack. The one for which we're adding the rule now knows about this but other don't.

Reply:
I think it makes total sense to document those implications. But I'd say that should be documented along with the instructions to _enable_ this relationship (probably in the [vsphere integration documentation](https://docs.newrelic.com/docs/infrastructure/host-integrations/host-integrations-list/vmware-vsphere-monitoring-integration/). After all, the label will be added in the integration configuration. Besides, adding this label is not different from adding any other [arbitrary label](https://docs.newrelic.com/docs/infrastructure/host-integrations/infrastructure-integrations-sdk/specifications/host-integrations-standard-configuration-format/#labels). WDYT?